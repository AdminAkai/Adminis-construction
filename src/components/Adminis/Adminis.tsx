import { FC } from 'react'

import { useHackerScramble } from 'src/hooks/stringHooks'

const Adminis: FC = () => {
  const word = useHackerScramble('Î»dminis')

  return <span>{word}</span>
}

export default Adminis
