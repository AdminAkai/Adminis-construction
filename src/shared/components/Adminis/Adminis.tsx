import { FC } from 'react'

import ScrambleText from '../ScrambleText'

const AdminisTitle = 'Î»dminis'

const Adminis: FC = () => (
  <span style={{ userSelect: 'none' }} aria-label='Adminis Title'>
    <ScrambleText text={AdminisTitle} />
  </span>
)

export default Adminis
