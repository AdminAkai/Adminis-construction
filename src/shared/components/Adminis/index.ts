import Adminis from './Adminis'

export default Adminis
