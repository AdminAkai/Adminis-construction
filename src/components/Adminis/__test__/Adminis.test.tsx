import { render, screen } from '@testing-library/react'
import renderer, { act } from 'react-test-renderer'
import '@testing-library/jest-dom'

import Adminis from '../Adminis.tsx'

const AdminisTitle = 'Î»dminis'

describe('Adminis tests', () => {
  beforeEach(() => {
    jest.useFakeTimers()
  })
  afterEach(() => {
    jest.clearAllTimers()
  })

  it('Should scramble then render Adminis', () => {
    const spy = jest.spyOn(global, 'setInterval')
    render(<Adminis />)
    const title = screen.getByLabelText('Adminis Title')
    expect(spy).toHaveBeenCalled()
    expect(title).toBeEmptyDOMElement()
    act(() => {
      for (let i = 0; i <= AdminisTitle.length; i++) {
        jest.advanceTimersByTime(310)
      }
    })
    expect(title).toHaveTextContent(AdminisTitle)
  })

  it('Adminis renders correctly', () => {
    const tree = renderer.create(<Adminis />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
