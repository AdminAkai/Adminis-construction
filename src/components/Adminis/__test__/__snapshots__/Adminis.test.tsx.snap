// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adminis tests Adminis renders correctly 1`] = `
<span
  aria-label="Adminis Title"
  style={
    {
      "userSelect": "none",
    }
  }
/>
`;
